src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentProperty.java

```diff
@@ -68,13 +76,26 @@ public SimpleElasticsearchPersistentProperty(Property property,
 		}
 	}
 
+	private String getAnnotatedFieldName() {
+
+		if (isAnnotationPresent(Field.class)) {
+
+			final Field annotation = findAnnotation(Field.class);
+			Assert.notNull(annotation, "findAnnotation() returned null on existing Annotation");
```

Comment:
You don't have to check it twice since `isAnnotationPresent` ensures that the annotation is present.

Reply:
IntelliJ warns me that it may be null, so I had this check (probably using too much Kotlin)